src/lib/activator.ts

```diff
@@ -0,0 +1,234 @@
+               });
+          };
+
+          module.exports.activate = async (event, context) => {
+               const ssm = new AWS.SSM('${region}');
+               const ag = new AWS.APIGateway('${region}');
+
+               let accessTokenRequest = (await ssm.getParameter({
+                    Name: '${ssmPrefix}/access_token',
+                    WithDecryption: false
+               }).promise().catch((err) => {
+                    return null;
+               }));
+
+
+               if (!accessTokenRequest) {
+                    console.log('Attempting Activation');
+                    let magentoUrl = (await ssm.getParameter({
+                         Name: '${ssmPrefix}/url',
+                         WithDecryption: false
+                    }).promise()).Parameter.Value;
+
+                    let serviceId = (await ssm.getParameter({
+                         Name: '${ssmPrefix}/service_id',
+                         WithDecryption: false
+                    }).promise()).Parameter.Value;
+
+                    let registrationToken = (await ssm.getParameter({
+                         Name: '${ssmPrefix}/registration_token',
+                         WithDecryption:true
+                    }).promise()).Parameter.Value;
+
+                    let interfaceContext = "";
+                    try {
+                         interfaceContext = (await ssm.getParameter({
+                              Name: '${ssmPrefix}/interface_context',
+                              WithDecryption: true
+                         }).promise()).Parameter.Value;
+                    } catch {
+                         console.debug('No interface context available');
+                    }
+
+                    let magentoToken = await activateService(magentoUrl, registrationToken, serviceId);
+                    if (magentoToken) {
+                         console.log('Service activated. Writing to SSM and creating service interface');
+                         await ssm.putParameter({
+                                   Name: '${ssmPrefix}/access_token',
+                                   Description: 'Written by @aligent/serverless-magento',
+                                   Value: magentoToken,
+                                   Type: 'SecureString'
+                              }).promise();
+
+                         await createServiceinterface(magentoUrl, registrationToken, serviceId, '${interfaceName}', '${interfaceAppUrl}', interfaceContext);
+                    } else {
+                         console.log('Service activation failed. Has it been approved?');
+                    }
+
+               } else {
+                    console.log('Service already activated.');
+               }
+
+          }`;
+
+          await fs.mkdir(handlerFolder, { recursive: true });
+          exec('npm init -y', { cwd: handlerFolder });
+          exec('npm install --save aws-sdk', { cwd: handlerFolder });
```

Comment:
If we make these dependencies of this module this wouldn't be needed right? 

Just not sure how I feel about executing npm like this

Reply:
No. These are deps of the activator handler not this module. So they get built into the temporary handler build folder.

Reply:
I believe we *could* add these as dependencies of the serverless application which is including this plugin however then we would have to advise anyone who uses the plugin that they will need these deps available.

Reply:
I.e the dependencies of `@aligent/serverless-magento` wont be included.